skyportal/facility_apis/_base.py

```diff
@@ -1,3 +1,38 @@
+from jsonschema_to_openapi.convert import convert
+from copy import deepcopy
+
+
+class _ListenerBase:
+
+    # subclasses should not modify this
+    @classmethod
+    def complete_schema(cls):
+        base = deepcopy(cls.schema)
```

Comment:
Can you add a docstring to explain what's happening in the code here?

Reply:
done

Reply:
Is this requested change pushed? I don't see a docstring

Reply:
not yet

Reply:
pushed

Reply:
The serialized capturing is not currently used anywhere and there is no API, so nothing to test at the moment. For now, it is purely record keeping in anticipation of possible future usages and for debugging. When we begin using the contents of the table, I will add tests.

Reply:
I added docstrings for all the functions. I request deferring the deployment of official docs for my forthcoming docs pr.